devices/homeetec.js

```diff
@@ -0,0 +1,360 @@
+
+        const result = {};
+        const key = Object.keys(SWITCH_DP).find((d) => SWITCH_DP[d] === dp);
+
+        if (key || false) {
+            result[key] = getOnState(value);
+        }
+        
+        meta.logger.debug(`homeetec_switch_datapoints (FZ) DEBUG, key: ${key}, value: ${value}, dpValue: ${JSON.stringify(dpValue)}, result: ${JSON.stringify(result)}`);
+
+        if (Object.keys(result).length > 0) {
+            return result;
+        }
+    },
+};
+
+tuya.fz.homeetec_cover = {
+    cluster: 'manuSpecificTuya',
+    type: ['commandDataResponse', 'commandDataReport', 'commandActiveStatusReport', 'commandActiveStatusReportAlt'],
+    options: [],
+    convert: (model, msg, publish, options, meta) => {
+        const dpValue = tuya.firstDpValue(msg, meta, 'homeetec_cover');
+        const dp = dpValue.dp;
+        const value = tuya.getDataValue(dpValue);
+        const modelMeta = model.meta || {};
+        const tuyaDatapoints = modelMeta.tuyaDatapoints || [];
+        const tuyaDatapoint = tuyaDatapoints.find((d) => d[0] === dp);
+
+        const result = {};
+            
+        const key = tuyaDatapoint ? tuyaDatapoint[1] : Object.keys(COVER_DP).find((d) => COVER_DP[d] === dp);
+
+        if (DP_OPTIONS[dp] || false) {
+            const options = DP_OPTIONS[dp];
+            const selectedOption = Object.keys(options).find((d) => options[d].toString() === value.toString());
+
+            if (selectedOption || false) {
+                result[key] = selectedOption;
+                meta.logger.debug(`homeetec_cover (FZ) DEBUG options, dp: ${dp}, key: ${key}, value: ${value}, options: ${JSON.stringify(options)}, dpValue: ${JSON.stringify(dpValue)}, result: ${JSON.stringify(result)}`);
+            } else {
+                meta.logger.debug(`homeetec_cover (FZ) Invalid value '${value}' provided for DP#${dp}, options: ${JSON.stringify(options)}`);
+            }
+
+        } else if (key === DP_KEYS.Position) {
+            const invert = tuya.isCoverInverted(meta.device.manufacturerName) ? !options.invert_cover : options.invert_cover;
+            const position = invert ? 100 - (value & 0xff) : value & 0xff;
+            result[key] = position;
+            meta.logger.debug(`homeetec_cover (FZ) DEBUG position, dp: ${dp}, key: ${key}, value: ${value}, dpValue: ${JSON.stringify(dpValue)}, result: ${JSON.stringify(result)}`);
+            
+        } else if ([DP_KEYS.Backlight, DP_KEYS.ChildLock, DP_KEYS.Switch, DP_KEYS.Switch1, DP_KEYS.Switch2].includes(key)) {
+            result[key] = getOnState(value);
+            meta.logger.debug(`homeetec_cover (FZ) DEBUG boolean, dp: ${dp}, key: ${key}, value: ${value}, dpValue: ${JSON.stringify(dpValue)}, result: ${JSON.stringify(result)}`);
+        }
+        
+        if (Object.keys(result).length > 0) {
+            return result;
+        }
+    },
+};
+
+const getDefintions = (model = null) => {
+    const definitions = [
+        {
```

Comment:
Could you check if this device works with the following config:

```js
{
        fingerprint: tuya.fingerprint('TS0001', ['_TZ3000_bmqxalil']),
        model: 'TS0001_switch_1_gang',
        vendor: 'TuYa',
        description: '1 gang switch',
        extend: tuya.extend.switch({powerOnBehavior2: true, backlightModeOffOn: true}),
        configure: async (device, coordinatorEndpoint, logger) => {
            await tuya.configureMagicPacket(device, coordinatorEndpoint, logger);
            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
        },
    }
```

Reply:
yes, that's extactly the code I added in homeetec.js,
once it's approved by you, I will move to tuya.js (wanted to avoid conflicts)

Reply:
Where can I see these changes you made? This PR still shows the old code.

Reply:
for some reason I forgot to commit that change, it's now available in `devices/homeetec.js`